app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java

```diff
@@ -676,23 +681,62 @@ protected void onActivityResult(int requestCode, int resultCode,
                     op = new PushOperation(localDir, activity).setCommand();
                     break;
 
+                case REQUEST_SYNC:
+                    op = new SyncOperation(localDir, activity).setCommands();
+                    break;
+
                 case GitOperation.GET_SSH_KEY_FROM_CLONE:
                     op = new CloneOperation(localDir, activity).setCommand(hostname);
                     break;
+
+                case SshApiSessionFactory.POST_SIGNATURE:
+                    return;
+
                 default:
-                    Log.e(TAG, "Operation not recognized : " + resultCode);
+                    Log.e(TAG, "Operation not recognized : " + operation);
                     setResult(RESULT_CANCELED);
                     finish();
                     return;
             }
 
-            try {
-                op.executeAfterAuthentication(connectionMode, settings.getString("git_remote_username", "git"), new File(getFilesDir() + "/.ssh_key"));
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            op.executeAfterAuthentication(connectionMode,
+                    settings.getString("git_remote_username", "git"),
+                    new File(getFilesDir() + "/.ssh_key"),
+                    identity);
+        } catch (Exception e) {
+            e.printStackTrace();
+            new AlertDialog.Builder(this).setMessage(e.getMessage()).show();
+        }
+    }
 
+    protected void onActivityResult(int requestCode, int resultCode,
+                                    Intent data) {
+
+        // In addition to the pre-operation-launch series of intents for OpenKeychain auth
+        // that will pass through here and back to launchGitOperation, there is one
+        // synchronous operation that happens /after/ the operation has been launched in the
+        // background thread - the actual signing of the SSH challenge. We pass through the
+        // completed signature to the ApiIdentity, which will be blocked in the other thread
+        // waiting for it.
+        if (requestCode == SshApiSessionFactory.POST_SIGNATURE && identity != null)
+            identity.postSignature(data);
+
+        if (resultCode == RESULT_CANCELED) {
+            setResult(RESULT_CANCELED);
+            finish();
+        } else if (resultCode == RESULT_OK) {
+            // If an operation has been re-queued via this mechanism, let the
+            // IdentityBuilder attempt to extract some updated state from the intent before
+            // trying to re-launch the operation.
+            if (identityBuilder != null) {
+                identityBuilder.consume(data);
+            }
+            launchGitOperation(requestCode);
         }
     }
 
+    @Override
```

Comment:
Useless override? You're not doing anything here

Reply:
Yep, this was left-over from debugging.